cogs/welcome/welcome.py

```diff
@@ -132,6 +188,139 @@ async def logServerLeave(self, leaveUser: discord.Member):
     async def welcome(self, ctx: Context):
         """Server welcome message settings."""
 
+    @checks.mod_or_permissions()
+    @welcome.command(name="channel")
+    async def welcomeChannelSet(self, ctx, channel: discord.TextChannel):
+        """
+        Set the welcome channel
+
+        Parameters:
+        -----------
+        channel: The text channel to set welcome's to
+        """
+        if not isinstance(channel, discord.TextChannel):
+            return
+
+        await self.config.guild(ctx.guild).KEY_WELCOME_CHANNEL.set(channel.id)
+        await self.config.guild(ctx.guild).KEY_WELCOME_CHANNEL_SET.set(True)
+        await ctx.send(f"Channel set to {channel}")
+
+        return
+
+    @checks.mod_or_permissions()
+    @welcome.command(name="add")
+    async def greetAdd(self, ctx, name):
+        """
+        Add a new greeting, please add name to the passed field. In the message, include '{USER}' for that to be replaced with a ping to the new user
+
+        Parameters:
+        -----------
+        name: name of the greeting
+        """
+
+        def check(message: discord.Message):
+            return message.author == ctx.message.author and message.channel == ctx.message.channel
+
+        await ctx.send("What would you like the greeting message to be?")
+        try:
+            greeting = await self.bot.wait_for("message", check=check, timeout=30.0)
+        except asyncio.TimeoutError:
+            await ctx.send("No response received, not setting anything!")
+            return
+
+        if len(greeting.content) > 2048:
+            await ctx.send("Your message is too long!")
+            return
+
+        greetings = await self.config.guild(ctx.guild).KEY_GREETINGS()
+
+        for greetingTuple in greetings:
```

Comment:
Same suggestion with dictionary, might be helpful here.

Reply:
using dictionary now